packages/server/graphql/public/typeDefs/toggleFavoriteTemplate.graphql

```diff
@@ -0,0 +1,20 @@
+extend type Mutation {
+  """
+  Add or remove the template to the user's favorite templates
+  """
+  toggleFavoriteTemplate(
+    """
+    The ID of the template to be toggled as a favorite
+    """
+    templateId: ID!
+  ): ToggleFavoriteTemplatePayload!
+}
+
+union ToggleFavoriteTemplatePayload = ErrorPayload | ToggleFavoriteTemplateSuccess
+
+type ToggleFavoriteTemplateSuccess {
+  """
+  The user who's favorite templates were updated
+  """
+  user: User!
```

Comment:
+1 Seems excessive to return the full user. I think it would be enough to return the updated template and favourite state.